app.py

```
@@ -0,0 +1,187 @@
+    return render_template("add_plant.html")
+
+
+""" View a plant """
+@app.route('/plants/<plant_id>', methods=['GET'])
+def view_plant(plant_id):
+    plant=mongo.db.plants.find_one({"_id": ObjectId(plant_id)})
+    return render_template('plant.html', plant=plant)
+
+
+""" Edit a plant """
+@app.route('/plant/edit/<plant_id>')
+def edit_plant(plant_id):
+    plant = mongo.db.plants.find_one({'_id': ObjectId(plant_id)})
+    # if request.method=='POST':
+    # form = request.form.to_dict()
+    # form.created_at = plant.created_at
+    # form.updated_at = datetime.datetime.now()
+    # form.created_by = input_created_by(plant.created_by, session['username'])
+    return render_template('edit_plant.html', plant=plant)
+
+
+@app.route('/update_plant/<plant_id>', methods=["POST"])
+def update_plant(plant_id):
+    plant = mongo.db.plants
+    plant.update({'_id': ObjectId(plant_id)},
+    {
+        'latin_name': request.form.get('latin_name'),
+        'common_name': request.form.get('common_name'),
+        'order': request.form.get('order'),
+        'family': request.form.get('family'),
+        'genus': request.form.get('genus'),
+        'preferred_lighting': request.form.get('preferred_lighting'),
+        'indoor_outdoor': request.form.get('indoor_outdoor'),
+        'water_frequency': request.form.get('water_frequency'),
+        'soil_type': request.form.get('soil_type'),
+        'additional_notes': request.form.get('additional_notes'),
+        'updated_at': datetime.datetime.now()
+    })
+    return redirect(url_for('view_plant', plant_id=plant_id))
+
+
+""" Delete a plant """
+@app.route('/delete_plant/<plant_id>')
+def delete_plant(plant_id):
+    mongo.db.plants.remove({'_id': ObjectId(plant_id)})
+    return redirect(url_for('view_plants'))
+
+""" Browse all plant genera """
+@app.route('/genera')
+def genera():
+    return render_template('genera.html', plants=mongo.db.plants.distinct({"genus": genus}), genus_name=genus)
```

This should throw an error as genus is undefined, doesn't it?
https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.distinct